cumulus_library/template_sql/statistics/count.sql.jinja

```diff
@@ -5,23 +5,45 @@
 {%- endmacro -%}
 
 {%- macro secondary_resource(fhir_resource) -%}
-{%- if fhir_resource in ('encounter', 'condition', 'documentreference', 'observation') -%}
+{%- if fhir_resource in ('encounter', 'condition') -%}
 encounter_ref
-{%- else -%}
-{# implicit null #}
-{%- endif -%}
-{%- endmacro -%}
-
-{%- macro tertiary_resource(fhir_resource) -%}
-{%- if fhir_resource == 'documentreference' -%}
+{%- elif fhir_resource == 'documentreference' -%}
 doc_ref
 {%- elif fhir_resource == 'observation'-%}
 observation_ref
 {%- else -%}
-{# implicit null #}
+{#- implicit null -#}
 {%- endif -%}
 {%- endmacro -%}
 
+{#- Moved extant queries to be secondary resource only, but leaving tertiary table
+infrastructure for future/moving resource specific logic to template generator -#}
+{%- macro tertiary_resource(fhir_resource) -%}
+{#- implicit null -#}
+{%- endmacro -%}
+
+{%- macro cols_delenated_list(table_cols, fhir_resource)-%}
+            {% for col in table_cols %}
+            "{{ column_or_alias(col) }}"
+            {{- comma_delineate(loop) }}
+            {%- endfor %}
+            {%- if fhir_resource in ('documentreference', 'observation' ) -%},
+            enc_class_display
+            {% endif %}
```

Comment:
This felt like an odd addition to this otherwise business-logic-neutral util method. As someone coming fresh to this code, this would surprise me. (here and in `coalesced_list`)

There are a couple places in this file where you iterate `table_cols` but don't include `enc_class_display` -- what is the different business-logic context there? Whatever it is, could you put that in the method name here, to clarify that this method also includes contextual logic stuff? (like `cols_for_cubing_as_a_delineated_list` or something)

Or is there a way to use `enc_class_display` in those other places, and then you could just have it be in the `table_cols` list and this method looks nice and lean?

(Also, I see in the final select, you don't use this method, because you need to prefix with `p.` -- maybe an optional `prefix` arg to the macro could help there.

And in `null_replacement` you don't use the coalesced version of this macro, because you are changing aliases at same time... hmm maybe that's harder to fix, unless you allow a `call` version of the macro)

Reply:
so i have this on my list of followon tasks - similar to how i moved `cond_display_code` out, i agree this would be good to move to the functions upstream that call the template generator method.

however - that's a larger structural change, because i then need to create a way to define a table join upstream (condition didn't have this problem since the table refactor included the foreign columns being referenced). I'm on the fence of how to do this atm (like, do i store some artifact that says which tables are joinable, and then leverage that, enforcing a two table only join? or do i provide a syntax like we do in the psm config, where you can specify arbitrary joins)?

Anyway, in the interest of not letting the perfect getting in the way of the good, I'm electing to defer this, since it doesn't structurally impact the final table output, just the way that you get there, but I'll handle this at some point soon (as referenced in the eaten comment).

Reply:
That's fine as a medium term goal. I guess my other request was "in the absence of refactoring the code, is there a name that makes more sense for this util method" -- something to indicate to a newbie to this code why this method includes a custom resource/column check.

Like, this is named like a generic utility method that will spit out column names. But you have lots of places where you don't use it because this method has additional business logic. So... some intern coming to this is going to be like "it's a bug that we don't use this utility method in these other looping cases" or something. I dunno what a good name is because it's not 100% to me what the semantic meaning of this business logic is compared to all the places you loop over `table_cols` _without_ including this logic. Which is part of my problem - the code flow is confusing and some comments or more descriptive names would help (absent any refactoring)